src/Illuminate/Support/Traits/Pipable.php

```diff
@@ -0,0 +1,51 @@
+<?php
+
+namespace Illuminate\Support\Traits;
+
+use Illuminate\Pipeline\Pipeline;
+use Throwable;
+
+trait Pipable
+{
+    /**
+     * @param array|string|callable $pipe
+     *
+     * @return $this
+     */
+    public function pipe($pipe)
+    {
+        return new class($this, $pipe)
+        {
+            private $pipable;
+
+            private $pipes;
+
+            public function __construct($pipable, $pipes)
+            {
+                $this->pipable = $pipable;
+                $this->pipes = $pipes;
+            }
+
+            public function __call($method, $params)
+            {
+                $pipeline = new Pipeline(app());
```

Comment:
app function cannot be called from the support component

Reply:
Yeah, I suspected that, what do you suggest to fix it?